measures/ApplyUpgrade/measure.rb

```diff
@@ -281,6 +279,34 @@ def run(model, runner, user_arguments)
         # Get measure name and arguments associated with the option
         options_measure_args = get_measure_args_from_option_names(lookup_file, [option_name], parameter_name, runner)
         options_measure_args[option_name].each do |measure_subdir, args_hash|
+          # Detect whether we are upgrading the heating system
+          args_hash.each do |arg, value|
+            if arg.include?('heating_system_type') || arg.include?('heating_system_fuel') || arg.include?('heating_system_heating_efficiency') || arg.include?('heating_system_fraction_heat_load_served')
+              upgrade_heating_system = true
+            end
+          end
+
+          # Detect whether we are upgrading the supplemental heating system
+          args_hash.each do |arg, value|
+            if arg.include?('heating_system_type_2') || arg.include?('heating_system_fuel_2') || arg.include?('heating_system_heating_efficiency_2') || arg.include?('heating_system_fraction_heat_load_served_2')
+              upgrade_suppy_heating_system = true
```

Comment:
Typo

Reply:
@shorowit Overall what do you think of this approach?

Reply:
I think it makes sense. Would be nice to have some tests that demonstrate it's working for different HVAC systems types. (Such a test would catch the above typo, for example! 😉)